data/src/main/java/com/android254/data/network/models/responses/GenericPaginatedResponse.kt

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2022 DroidconKE
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.android254.data.network.models.responses
+
+import kotlinx.serialization.Serializable
+
+@Serializable
+data class GenericPaginatedResponse<ResponseData>(
```

Comment:
Why is this named Generic?

Reply:
> Why is this named Generic?

Just in case there are other types of paginated responses.
I could rename it to PaginatedResponse

Reply:
We go with the name of the data being fetched , in this case sessions

Reply:
> e



> Why is this named Generic?

I finally found out why I named it like so. It's a class that takes a generic `ResponseData` object, so in this context it will be the `SessionApiModel` this way we do not need to create 2 classes for each response  `SessionsResponse` (for a collection/list of sessions) and `SessionResponse` (for a session), so we can reuse this `PaginatedResponse` for many other similar responses that have a paginated collection/list of objects.

Reply:
Aah okay i'd prefere removing the Generic word